.github/workflows/build.yml

```diff
@@ -11,51 +15,57 @@ jobs:
   lint:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions/setup-python@v1
-        with:
-          python-version: 3.7
-      - name: Cache pip test requirements
-        uses: actions/cache@v1
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
         with:
-          path: ${{ env.PIP_CACHE_DIR }}
-          key: "${{ runner.os }}-pip-test-\
-            ${{ hashFiles('**/requirements-test.txt') }}"
-          restore-keys: |
-            ${{ runner.os }}-pip-test-
-            ${{ runner.os }}-pip-
-      - name: Cache pre-commit hooks
-        uses: actions/cache@v1
+          python-version: 3.8
+      - name: Store installed Python version
+        run: |
+          echo "::set-env name=PY_VERSION::"\
+          "$(python -c "import platform;print(platform.python_version())")"
+      - name: Cache linting environments
+        uses: actions/cache@v2
         with:
-          path: ${{ env.PRE_COMMIT_CACHE_DIR }}
-          key: "${{ runner.os }}-pre-commit-\
+          path: |
+            ${{ env.PIP_CACHE_DIR }}
+            ${{ env.PRE_COMMIT_CACHE_DIR }}
+          key: "lint-${{ runner.os }}-py${{ env.PY_VERSION }}-\
+            ${{ hashFiles('**/requirements-test.txt') }}-\
             ${{ hashFiles('**/.pre-commit-config.yaml') }}"
+          restore-keys: |
+            lint-${{ runner.os }}-py${{ env.PY_VERSION }}-
+            lint-${{ runner.os }}-
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
-          pip install --upgrade -r requirements-test.txt
-      - name: Run linters on all files
+          pip install --upgrade --requirement requirements-test.txt
+      - name: Set up pre-commit hook environments
+        run: pre-commit install-hooks
+      - name: Run pre-commit on all files
         run: pre-commit run --all-files
   test:
     runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
     steps:
-      - uses: actions/checkout@v1
-      - uses: actions/setup-python@v1
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
         with:
-          python-version: 3.7
-      - name: Cache pip test requirements
-        uses: actions/cache@v1
+          python-version: ${{ matrix.python-version }}
+      - name: Cache testing environments
+        uses: actions/cache@v2
         with:
           path: ${{ env.PIP_CACHE_DIR }}
-          key: "${{ runner.os }}-pip-test-\
+          key: "test-${{ runner.os }}-py${{ matrix.python-version }}-\
```

Comment:
Please add `requirements.txt` to the cache key below, similar to what we did in https://github.com/cisagov/guacamole-composition/pull/12/commits/15cf9ca26e5bcff351e078d4ba17d603349a5b3f.

Reply:
The skeleton this is pulling from does not yet have the update to the `lint` job for the cache key. The PR will percolate down with both key updates if we wanted to wait for that PR to generate. If not, then the `lint` job also needs to be updated:
https://github.com/cisagov/assessment-data-import-lambda/blob/c6a525ec4653a6ba07572da322a0b03220d4771e/.github/workflows/build.yml#L32-L34

Reply:
Ah, you are correct @mcdonnnj - we can wait until [`skeleton-aws-lambda`](https://github.com/cisagov/skeleton-aws-lambda) gets this change; no need to fix it here now.

Reply:
Resolving because this will be inherited.